pkg/diskmaker/discovery/monitor.go

```diff
@@ -0,0 +1,123 @@
+package discovery
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/klog"
+)
+
+// Monitors udev for block device changes, and collapses these events such that
+// only one event is emitted per period in order to deal with flapping.
+func udevBlockMonitor(c chan string, period time.Duration) {
+	defer close(c)
+	var udevFilter []string
+
+	// return any add or remove events, but none that match device mapper
+	// events. string matching is case-insensitive
+	events := make(chan string)
+
+	// regex to ignore the list of devices where udev events should not
+	// trigger continous discovery
+	discoverUdev := "(?i)dm-[0-9]+,(?i)rbd[0-9]+,(?i)nbd[0-9]+"
+
+	udevFilter = strings.Split(discoverUdev, ",")
+	klog.Infof("using the regular expressions %q", udevFilter)
+
+	go rawUdevBlockMonitor(events,
+		[]string{"(?i)add", "(?i)remove"},
+		udevFilter)
+
+	for {
+		event, ok := <-events
+		if !ok {
+			return
+		}
+		timeout := time.NewTimer(period)
```

Comment:
So as I understand this - this code here tries to serialize udev events that were received in 5second interval and send only one event to discover goroutine, so as each udev event does not trigger a full device discovery?